app/components/EmojiCreateDialog.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from "react";
+import { useDropzone } from "react-dropzone";
+import { useTranslation } from "react-i18next";
+import { toast } from "sonner";
+import styled from "styled-components";
+import { s } from "@shared/styles";
+import { AttachmentPreset } from "@shared/types";
+import ConfirmationDialog from "~/components/ConfirmationDialog";
+import Flex from "~/components/Flex";
+import Input from "~/components/Input";
+import Text from "~/components/Text";
+import useStores from "~/hooks/useStores";
+import { uploadFile } from "~/utils/files";
+import { compressImage } from "~/utils/compressImage";
+import Logger from "~/utils/Logger";
+
+type Props = {
+  onSubmit: () => void;
+};
+
+export function EmojiCreateDialog({ onSubmit }: Props) {
+  const { t } = useTranslation();
+  const { emojis } = useStores();
+  const [name, setName] = React.useState("");
+  const [file, setFile] = React.useState<File | null>(null);
+  const [isUploading, setIsUploading] = React.useState(false);
+
+  const onDrop = React.useCallback((acceptedFiles: File[]) => {
+    if (acceptedFiles.length > 0) {
+      setFile(acceptedFiles[0]);
+    }
+  }, []);
+
+  const { getRootProps, getInputProps, isDragActive } = useDropzone({
+    onDropAccepted: onDrop,
+    accept: [".png", ".jpg", ".jpeg", ".gif", ".webp"],
+    maxFiles: 1,
+    maxSize: 1024 * 1024, // 1MB
+  });
+
+  const handleSubmit = async () => {
+    if (!name.trim()) {
+      toast.error(t("Please enter a name for the emoji"));
+      return;
+    }
+
+    if (!file) {
+      toast.error(t("Please select an image file"));
+      return;
+    }
+
+    setIsUploading(true);
+    try {
+      const compressed = await compressImage(file, {
+        maxHeight: 64,
+        maxWidth: 64,
+      });
+
+      const attachment = await uploadFile(compressed, {
+        name: file.name,
+        preset: AttachmentPreset.DocumentAttachment,
+      });
+
+      await emojis.create({
+        name: name.trim(),
+        url: attachment.url,
+      });
+
+      // to do: update with optimistic state
+      toast.success(t("Emoji created successfully"));
+      onSubmit();
+    } catch (error) {
+      toast.error(t("Failed to create emoji"));
+      Logger.error("Failed to create emoji", error);
+    } finally {
+      setIsUploading(false);
+    }
+  };
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value
+      .toLowerCase()
+      .replace(/[^a-z0-9_]/g, "")
```

Comment:
Lets move this to `shared/validations` and share it with the backend model field validation